neo/Network/P2P/Message.cs

```diff
@@ -1,79 +1,139 @@
-﻿using Neo.Cryptography;
+﻿using System;
+using System.IO;
+using Akka.IO;
+using Neo.Cryptography;
 using Neo.IO;
 using Neo.Network.P2P.Payloads;
-using System;
-using System.IO;
 
 namespace Neo.Network.P2P
 {
     public class Message : ISerializable
     {
-        public const int HeaderSize = sizeof(uint) + 12 + sizeof(int) + sizeof(uint);
         public const int PayloadMaxSize = 0x02000000;
+        public const int CompressionMinSize = 180;
+        public const int CompressionThreshold = 100;
 
-        public static readonly uint Magic = ProtocolSettings.Default.Magic;
-        public string Command;
-        public uint Checksum;
```

Comment:
What happened to the `CheckSum`?

Reply:
Gone :) tcp have checksum, is faster

Reply:
I think we still add the `CheckSum` because we also need to support UDP or other protocols.

Reply:
I think we still add the `CheckSum` because we also need to support UDP or other protocols.

Reply:
Maybe we can add a flag for Checksum, and take advantage of the benefits of every transport layer. So we can use it, or not, according to this flag

Reply:
Add a new flag in `MessageFlags`?

Reply:
No, inside the current one, Checksum and Compression in the same enum

Reply:
Maybe we should use a `short` for this enum, instead of one byte?

Reply:
What else will be put in `MessageFlags`? A byte may be enough.

Reply:
Check this change https://github.com/neo-project/neo/pull/710/commits/7d55288d500abd49169bcc5fb3521ffb33b6e330 maybe we need to find a faster CRC method, but i think that this should be the way